src/ucp/wireup/wireup.c

```diff
@@ -204,33 +252,49 @@ ucs_status_t ucp_select_transport(ucp_ep_h ep,
         for (ae = address_list; ae < address_list + address_count; ++ae) {
```

Comment:
if a higher score already exists and found=1, then no need to check if remote peer is reachable for this transport's device. it won't be chosen anyway...